src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,270 @@
+                return snapshot.Storages.TryGet(CreateTokenKey(innerKey)).GetInteroperable<TState>();
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.InteropInterface, CallFlags.None, ParameterTypes = new[] { ContractParameterType.ByteArray }, ParameterNames = new[] { "tokenId" })]
+        public StackItem OwnerOf(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenId = args[0].GetSpan().ToArray();
+            return new InteropInterface(OwnerOf(engine.Snapshot, tokenId));
+        }
+
+        public virtual IEnumerator<UInt160> OwnerOf(StoreView snapshot, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_Ownership, innerKey).ToArray()).Select(p =>
+            {
+                return new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_08000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.Hash160, ContractParameterType.Hash160, ContractParameterType.Integer, ContractParameterType.ByteArray }, ParameterNames = new[] { "from", "to", "amount", "tokenId" })]
+        public virtual StackItem Transfer(ApplicationEngine engine, Array args)
+        {
+            UInt160 from = new UInt160(args[0].GetSpan());
+            UInt160 to = new UInt160(args[1].GetSpan());
+            BigInteger amount = args[2].GetBigInteger();
+            byte[] tokenId = args[3].GetSpan().ToArray();
+            return Transfer(engine, from, to, amount, tokenId);
+        }
+
+        public virtual bool Transfer(ApplicationEngine engine, UInt160 from, UInt160 to, BigInteger amount, byte[] tokenId)
+        {
+            if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, from)) return false;
+
+            var storages = engine.Snapshot.Storages;
+            ContractState contract_to = engine.Snapshot.Contracts.TryGet(to);
+            if (contract_to?.Payable == false) return false;
+            if (!amount.IsZero && !from.Equals(to))
+            {
+                UInt160 innerKey = GetInnerKey(tokenId);
+                StorageKey fromKey = CreateOwnershipKey(from, innerKey);
+                UState fromBalance = storages.GetAndChange(fromKey)?.GetInteroperable<UState>();
+                if (fromBalance is null) return false;
+                if (fromBalance.Balance < amount) return false;
+                if (fromBalance.Balance == amount)
+                {
+                    storages.Delete(fromKey);
+                    storages.Delete(CreateOwnershipKey(innerKey, from));
+                }
+                fromBalance.Balance -= amount;
```

Comment:
We alter the object after `Delete` the entry

Reply:
Yep, seems like this should be inside an `else` branch